examples/md-flexible/src/domainDecomposition/DomainTools.cpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file DomainTools.cpp
+ * @author J. Körner
+ * @date 13.05.2021
+ */
+#include "DomainTools.h"
+
+#include <cmath>
+#include <list>
+
+namespace DomainTools {
+bool isInsideDomain(const std::vector<double> &coordinates, std::vector<double> &boxMin, std::vector<double> &boxMax) {
+  bool isInsideLocalDomain = true;
+  for (int i = 0; i < coordinates.size(); ++i) {
+    if (!isInsideLocalDomain) {
+      break;
+    }
+    isInsideLocalDomain = coordinates[i] >= boxMin[i] && coordinates[i] < boxMax[i];
+  }
+  return isInsideLocalDomain;
+}
+
+bool isInsideDomain(const std::array<double, 3> &coordinates, std::array<double, 3> &boxMin,
+                    std::array<double, 3> &boxMax) {
+  bool isInsideLocalDomain = true;
+  for (int i = 0; i < coordinates.size(); ++i) {
+    if (!isInsideLocalDomain) {
+      break;
+    }
+    isInsideLocalDomain = coordinates[i] >= boxMin[i] && coordinates[i] < boxMax[i];
+  }
+  return isInsideLocalDomain;
+}
```

Comment:
```suggestion
template<class T>
bool isInsideDomain(const T &coordinates, std::array<double, 3> &boxMin,
                    std::array<double, 3> &boxMax) {
  bool isInsideLocalDomain = true;
  for (int i = 0; i < coordinates.size(); ++i) {
    if (!isInsideLocalDomain) {
      break;
    }
    isInsideLocalDomain = coordinates[i] >= boxMin[i] and coordinates[i] < boxMax[i];
  }
  return isInsideLocalDomain;
}
```

Similar with `getDistanceToDomain()` to avoid code duplication. However, this then has to go to the header.

Reply:
On second thought: Why do you need the vector variant anyways?

Reply:
Again this is because of me having multiple dimensions in mind. Therefore I am using vectors where arrays might be sufficient. I guess I will limit the whole logic to 3 dimensions to avoid such helper functions and other problems.